attestation-service/src/verifier/cca/mod.rs

```diff
@@ -0,0 +1,263 @@
+        },
+        {
+            1: "M2",
+            5: h'07060504030201000f0e0d0c0b0a090817161514131211101f1e1d1c1b1a1918',
+            4: "1.2.3",
+            2: h'07060504030201000f0e0d0c0b0a090817161514131211101f1e1d1c1b1a1918',
+        },
+        {
+            1: "M3",
+            5: h'07060504030201000f0e0d0c0b0a090817161514131211101f1e1d1c1b1a1918',
+            4: "1",
+            2: h'07060504030201000f0e0d0c0b0a090817161514131211101f1e1d1c1b1a1918',
+        },
+    ],
+    2400_1: "whatever.com",
+}
+{
+    10: h'00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+    44235_1: h'00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+    44237_1: h'0476f988091be585ed41801aecfab858548c63057e16b0e676120bbd0d2f9c29e056c5d41a0130eb9c21517899dc23146b28e1b062bd3ea4b315fd219f1cbb528cb6e74ca49be16773734f61a1ca61031b2bbf3d918f2f94ffc4228e50919544ae',
+    44236_1: "sha-256",
+    44240_1: "sha-256",
+    44238_1: h'75a1fbc79a7d20a5ff843b914dfd8093d40cd07dd633401c8c42d697be224801',
+    44239_1: [
+        h'0000000000000000000000000000000000000000000000000000000000000000',
+        h'0000000000000000000000000000000000000000000000000000000000000000',
+        h'0000000000000000000000000000000000000000000000000000000000000000',
+        h'0000000000000000000000000000000000000000000000000000000000000000',
+    ],
+}
+*/
+fn parse_cca_token(token: Vec<u8>) -> Result<Evidence> {
+    let evidence = Evidence {
+        cca_realm_delegated_token: RealmToken {
+            cca_realm_initial_measurement: "".to_string(),
+        },
+    };
+
+    // NOTE: For some corner case, the date cannot be parsed to token correctly but the token
+    // can be validated successfully by the CCA verifier e.g. `Err` value: Todo("Remaining bytes (00)")'
+    // Instead of throwing an error, just print it out in this case.
+    let mut di = match cbor_diag::parse_bytes(token) {
+        Ok(di) => di,
+        Err(err) => {
+            info!("Error: {:?}", err);
+            return Ok(evidence);
+        }
+    };
+
+    if let cbor_diag::DataItem::Tag {
+        tag: _,
+        bitwidth: _,
+        value,
+    } = di
+    {
+        di = *value;
+    }
+
+    if let cbor_diag::DataItem::Map { data, .. } = di {
+        for item in data {
+            if let cbor_diag::DataItem::ByteString(t) = item.1 {
+                let tok = cbor_diag::parse_bytes(t.data).unwrap();
+                if let cbor_diag::DataItem::Tag {
+                    tag: _,
+                    bitwidth: _,
+                    value,
+                } = tok
+                {
+                    if let cbor_diag::DataItem::Array { data, bitwidth: _ } = *value {
+                        if let cbor_diag::DataItem::ByteString(cose) = data.get(2).unwrap() {
```

Comment:
```suggestion
                        if let cbor_diag::DataItem::ByteString(cose) = data.get(2).ok_or_else(|| anyhow!("...error info")) {
```

Reply:
hmm... this won't compile, I will come back to this to see how to fix it.

Reply:
@Xynnn007 I thought about the `unwrap_or_else` where we can either provide a default for else branch or panic it anyway, 

- default value: there is no default value available here
- panic will be the same as the `unwrap`

I also thought about `unwrap_or_default` or `ok_or_else`, both of them doesn't apply for the case here.

Reply:
Can we leave it as is or any suggestion?

Reply:
sorry, I test on my machine. there should add an extra `?` after `ok_or_else`. s.t.
```
if let cbor_diag::DataItem::ByteString(cose) = data.get(2).ok_or_else(|| anyhow!("...error info"))? {
```



> @Xynnn007 I thought about the `unwrap_or_else` where we can either provide a default for else branch or panic it anyway,
> 
> * default value: there is no default value available here
> * panic will be the same as the `unwrap`
> 
> I also thought about `unwrap_or_default` or `ok_or_else`, both of them doesn't apply for the case here.

Panic is not encouraged in this project. We should try to raise error to the upper caller to have an elegant error handling. Unexpected `unwrap` will make it unstable when using attestation-service.

Reply:
thanks for help!